src/pybind/mgr/cephadm/agent.py

```diff
@@ -130,22 +146,112 @@ def shutdown(self) -> None:
         self.cherrypy_shutdown_event.set()
 
 
-class Root:
-    exposed = True
+class Root(object):
+
+    # collapse everything to '/'
+    def _cp_dispatch(self, vpath: str) -> 'Root':
+        cherrypy.request.path = ''
+        return self
 
     def __init__(self, mgr: "CephadmOrchestrator"):
         self.mgr = mgr
-        self.data = HostData(self.mgr)
 
-    def GET(self) -> str:
+    @cherrypy.expose
+    def index(self) -> str:
         return '''<!DOCTYPE html>
 <html>
 <head><title>Cephadm HTTP Endpoint</title></head>
 <body>
-<p>Cephadm HTTP Endpoint is up and running</p>
+<p>Cephadm Service discovery Endpoints</p>
+<p><a href='sd-config?service=prometheus'>Prometheus http sd-config</a></p>
+<p><a href='sd-config?service=alert-manager'>Prometheus http sd-config</a></p>
```

Comment:
should this be alertmanager instead of alert-manager so it matches with the actual service name?

Reply:
fixed: let's go witht **alertmanager**

Reply:
Actually, thinking about it, I'm not sure what this is intended to be for? I think if anything, it should tie in to the receiver definition pointing to the dashboard's alertmanager API endpoint - shouldn't it?e.g.
```
- name: 'ceph-dashboard'
  webhook_configs:
  - url: 'https://<mgr1_ip>:<port>/api/prometheus_receiver'
  - url: 'https://<mgr2_ip>:<port>/api/prometheus_receiver'
```
What I see at the moment is this; 
[{"targets": ["10.70.39.224:9093"], "labels": {}}]
And we don't scrape the alertmanager instance.

Maybe consider pulling this out of the sd-config group and putting something similar to the rules endpoint.

Reply:
@rkachach apologies for the delay!
The service discovery endpoints all relate to allowing an external prometheus server to pick up the endpoints for scrape-configs right? So my question relates to how is the alertmanager info useful?  If you have external monitoring solution, you pick up the scrape jobs, and can pick up and apply the alert rules. The alert rules will then fire within the external monitoring solution. I just don't understand why the alertmanager endpoint is needed here?

Reply:
@pcuzner I see your point, wouldn't that be the case also for node-exporter (for external monitoring solutions)? 
because in both cases we just point to the external daemons (alert-manager, node-exporter).

If yes, then maybe we need just to publish EP for:
- mgr-prometheus
- ha-proxy

Reply:
@rkachach Sorry if I've confused things - I think I got stuck on value to the external monitoring solution, ignoring the value the sd_config plays locally to simplify the config and reconfig task.

You are right to include the alertmanager as a sd_config target. It needs to be defined in the alerting: section of prometheus.yml to define where alerts are sent by our prometheus server AND defined in the scrape_configs section to gather metrics relating to alertmanager state. We don't actually use any of the alertmanager metrics currently, but we should think about using them in future for alertmanager health, peer status etc

So to be clear having an endpoint for node-exporter, mgr/prometheus, alertmanager and haproxy are needed today - and in future we can add ceph-exporter to that list.

I think my original point about an alertmanager webhook_config endpoint still holds true for integration with an external prometheus and alertmanager config. For example, lets include an endpoint that passes back the webhook urls for the dashboard api to receive events from an external alertmanager. That makes it easier for the monitoring team to ensure that when they add the webhook to their alertmanager it is 100% accurate - which means the dashboard will receive push events from the external alertmanager cluster (they'll still need to define appropriate routing with a "matchers" clause, but I think this would be helpful. As I mentioned before this would be another endpoint, in a similar vein to the alert rules.

Make sense?

Reply:
I think we are on the same page :)

Once these changes are merged the next step would be adding support to use the service discovery to simplify the config/update of our deployed daemons (Prometheus stack, etc). It would be good also to test these new changes with a real external monitoring stack to see if we are good to go and to detect any missing functionality.